src/compiler/Stage/Desugar.elm

```diff
@@ -262,11 +235,58 @@ qualifiedVarInImportedModule modules maybeModuleName varName =
         |> Maybe.withDefault Nothing
 
 
-qualifiedVarInAliasedModule : Modules Frontend.Expr -> Module Frontend.Expr -> Maybe ModuleName -> VarName -> Maybe ModuleName
+qualifiedVarInAliasedModule : Modules Frontend.LocatedExpr -> Module Frontend.LocatedExpr -> Maybe ModuleName -> VarName -> Maybe ModuleName
 qualifiedVarInAliasedModule modules thisModule maybeModuleName varName =
     let
         unaliasedModuleName =
             Maybe.andThen (Common.unalias thisModule) maybeModuleName
     in
     -- Reusing the existing functionality. TODO is this a good idea?
     qualifiedVarInImportedModule modules unaliasedModuleName varName
+
+
+locatedResultReturn : Frontend.LocatedExpr -> Canonical.Expr -> Result DesugarError Canonical.LocatedExpr
+locatedResultReturn located expr =
+    Ok
+        (Located.map
```

Comment:
These wrappers call for some non-map helper like

```elm
Located.replaceWith : b -> Located a -> Located b
Located.replaceWith new (Located region _) =
  Located region new
```

Then this can be nicer:
```elm
Ok <| Located.replaceWith expr located
```
similarly for the other functions